app/frontend/src/features/BugReport.test.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+  within,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import mediaQuery from "css-mediaquery";
+import { report } from "process";
+
+import { service } from "../service";
+import wrapper from "../test/hookWrapper";
+import { addDefaultUser, MockedService } from "../test/utils";
+import BugReport from "./BugReport";
+
+const reportBugMock = service.bugs.reportBug as MockedService<
+  typeof service.bugs.reportBug
+>;
+
+afterEach(() => jest.restoreAllMocks);
+
+async function fillInAndSubmitBugReport(
+  subjectFieldLabel: string,
+  descriptionFieldLabel: string,
+  stepsFieldLabel: string = "",
+  resultsFieldLabel: string = ""
+) {
+  const subjectField = await screen.findByLabelText(subjectFieldLabel);
+  const descriptionField = await screen.findByLabelText(descriptionFieldLabel);
+  const stepsField = screen.queryByLabelText(stepsFieldLabel);
+  const resultsField = screen.queryByLabelText(resultsFieldLabel);
```

Comment:
`findBy*` throws an error if the text field can't be found, whereas `queryBy*` returns null. It's only needed because I'm trying to make this reusable and for some test cases, I'm not filling in the `stepsField` and `resultsField` on purpose just to see if submit still works since they are optional.